domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -3281,6 +3281,52 @@ void getSQLConnection(int queryTimeout, Handler<AsyncResult<SQLConnection>> hand
     });
   }
 
+  /**
+   * Execute the given function within a transaction.
+   * <p>Similar to {@link PgPool#withTransaction(Function)}
+   * <ul>
+   *   <li>The connection is automatically closed in all cases when the function exits.</li>
+   *   <li>The transaction is automatically committed if the function returns a succeeded Future.
+   *   The transaction is automatically roll-backed if the function returns a failed Future or throws a Throwable.</li>
+   *   <li>The method returns a succeeded Future if the commit is successful, otherwise a failed Future.</li>
+   * </ul>
+   *
+   * @param function code to execute
+   * @param <T> type
+   * @return result handler
+   */
+  public <T> Future<T> withTransaction(Function<PgConnection, Future<T>> function) {
+    return getConnection()
+      .flatMap(conn -> conn
+        .begin()
+        .flatMap(tx -> function
+          .apply(conn)
+          .compose(
+            res -> tx
+              .commit()
+              .flatMap(v -> Future.succeededFuture(res)),
+            err -> tx
+              .rollback()
+              .compose(v -> Future.failedFuture(err), failure -> Future.failedFuture(err))))
+        .onComplete(ar -> conn.close()));
+  }
+
+  /**
+   * Get a connection from the pool and execute the given function.
+   * <p>Similar to {@link PgPool#withConnection(Function)}
+   * <ul>
+   *   <li>The connection is automatically closed in all cases when the function exits.</li>
+   *   <li>The method returns a succeeded Future if the function is successful, otherwise a failed Future.</li>
```

Comment:
```suggestion
   *   <li>The method returns the Future returned by the function, or a failed Future with the Throwable thrown by the function.</li>
```
Mention the exception handling.

Reply:
ok